lib/baseFactory.js

```diff
@@ -145,6 +148,36 @@ class BaseFactory {
                 return result;
             });
     }
+
+    /**
+     * Get an instance of a Factory
+     * @method getInstance
+     * @param  {Object}     ClassDef            Class definition of a factory
+     * @param  {Object}     instance            The current instance of a factory
+     * @param  {Object}     config
+     * @param  {Datastore}  config.datastore    A datastore instance
+     * @param  {Datastore}  config.scmPlugin    A scm plugin instance
+     * @return {Factory}
+     */
+    static getInstance(ClassDef, instance, config) {
+        let inst = instance;
+
+        if (!instance) {
```

Comment:
Minor thing, but should this be `!inst` just for consistency of only using inst vs instance?